src/snowflake/connector/auth_webbrowser.py

```diff
@@ -122,23 +118,17 @@ def authenticate(self, authenticator, service_name, account, user, password):
 
             logger.debug("step 2: open a browser")
             if not self._webbrowser.open_new(sso_url):
-                logger.error(
-                    "Unable to open a browser in this environment.", exc_info=True
-                )
                 print(
                     "We were unable to open a browser window for you, "
-                    f"please open the following url manually: {sso_url}"
+                    "please open the following url manually then paste the "
+                    "URL you are redirected to into the terminal."
                 )
-                self.handle_failure(
-                    {
-                        "code": ER_UNABLE_TO_OPEN_BROWSER,
-                        "message": "Unable to open a browser in this environment.",
-                    }
-                )
-                return  # required for test case
```

Comment:
I'm sure that this broke some tests

Reply:
@sfc-gh-mkeller thanks will take a look (and also rebase)

Reply:
@sfc-gh-mkeller you were absolutely right, I've added some additional test edge cases.
I've added them to the "fail_webbrowser" test case using `pytest.mark.parametrize`. Let me know if you'd like me to use a different approach.